### PR TITLE
fix: key compression always

### PR DESCRIPTION
**Pull request by andrei-stoian-zama**
Enable key compression for all models

### CODE REVIEW

File: src/concrete/ml/sklearn/linear_model.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def _get_training_quantized_module(
         # Enable the underlying FHE circuit to be composed with itself
         # This feature is used in order to be able to iterate in the clear n times without having
         # to encrypt/decrypt the weight/bias values between each loop
-        configuration = Configuration(composable=True)
+        configuration = Configuration(composable=True, compress_evaluation_keys=True)
```
Comment (andrei-stoian-zama) [inline src/concrete/ml/sklearn/linear_model.py:350]: compress pbs keys for training

File: tests/torch/test_hybrid_converter.py [hunk 2]
```diff
@@ -40,6 +40,8 @@ def run_hybrid_llm_test(
     # Multi-parameter strategy is used in order to speed-up the FHE executions
     configuration = Configuration(
         single_precision=False,
+        compress_input_ciphertexts=True,
```
Comment (andrei-stoian-zama) [inline tests/torch/test_hybrid_converter.py:43]: these circuits can be fully leveled or not, always good to test with default compilation parameters

File: docker/release_resources/sanity_check.py [hunk 3]
```diff
@@ -120,6 +122,8 @@ def function_to_compile(x):
         enable_unsafe_features=is_fast,
         use_insecure_key_cache=is_fast,
         insecure_key_cache_location=keyring_dir_as_str,
+        compress_input_ciphertexts=True,
```
Comment (andrei-stoian-zama) [inline docker/release_resources/sanity_check.py:125]: sanity checks should run with default compile parameters

File: conftest.py [hunk 4]
```diff
@@ -173,6 +174,7 @@ def simulation_configuration():
         fhe_simulation=True,
         fhe_execution=False,
         compress_input_ciphertexts=os.environ.get("USE_INPUT_COMPRESSION", "1") == "1",
+        compress_eval_keys=True,
```
Comment (andrei-stoian-zama) [inline conftest.py:177]: tests should use the default for compressed keys, which is True

**Review by RomanBredehoft — CHANGES_REQUESTED** [review]
imo we shouldn't force it like this but do like for input compression, meaning we use env variables. The main reason is that forcing it like this makes it impossible to disable it for tests or other things through input `configuration` objects. Basically this is what I reverted for fhe/simu exec in https://github.com/zama-ai/concrete-ml/pull/724 (which is going to conflict a bit with your PR btw) 

additionally we probably want to add tests to make sure that the compression properly works, like we did for input compression

File: src/concrete/ml/sklearn/linear_model.py (hunk 1 above)
Comment (RomanBredehoft) [reply src/concrete/ml/sklearn/linear_model.py:350]: which makes me thinkg, it should be using `enable_input_compression` as well here !

File: tests/torch/test_hybrid_converter.py (hunk 2 above)
Comment (RomanBredehoft) [reply tests/torch/test_hybrid_converter.py:43]: so why not also enabling key compression ?

**Review by RomanBredehoft — APPROVED** [review]
thanks !! I think we could add a few small things in a follow up PR

File: tests/torch/test_hybrid_converter.py (hunk 2 above)
Comment (andrei-stoian-zama) [reply tests/torch/test_hybrid_converter.py:43]: it is enabled by the QM compilation (the env variable defaults to 1)

File: src/concrete/ml/sklearn/linear_model.py (hunk 1 above)
Comment (andrei-stoian-zama) [reply src/concrete/ml/sklearn/linear_model.py:350]: yes, you're right, same as for DF which was not enabled before

File: tests/torch/test_hybrid_converter.py (hunk 2 above)
Comment (RomanBredehoft) [reply tests/torch/test_hybrid_converter.py:43]: so is `compress_input_ciphertexts` then right ? imo either we put both here or we revert this small change 🤔

File: src/concrete/ml/sklearn/linear_model.py (hunk 1 above)
Comment (RomanBredehoft) [reply src/concrete/ml/sklearn/linear_model.py:350]: actually same as below, we should probably revert this change : it's emabled by QM's compilation anyway

**Pull request merged**